etl/kpinetwork_analizer.py

```diff
@@ -338,6 +561,98 @@ def get_dataframes_from_lists(data):
     )
 
 
+def get_updating_dataframes_from_lists(data):
+    (companies, periods, scenarios, metrics, scenario_metrics, currencies) = data
+
+    df_companies = spark.createDataFrame(companies, schema=company_schema)
+    df_time_periods = spark.createDataFrame(periods, schema=time_period_schema)
+    df_time_periods = dataframe_cast_date_type(df_time_periods, ["start_at", "end_at"])
+    df_scenarios = spark.createDataFrame(scenarios, schema=scenario_schema)
+    df_metrics = spark.createDataFrame(metrics, schema=metric_schema)
+    df_currencies = spark.createDataFrame(currencies, schema=currency_schema)
+    df_scenario_metrics = spark.createDataFrame(
+        scenario_metrics, schema=scenario_metric_schema
+    )
+
+    return (
+        df_companies,
+        df_time_periods,
+        df_scenarios,
+        df_metrics,
+        df_scenario_metrics,
+        df_currencies,
+    )
+
+
+def call_lambda_to_update(env, data):
+    name = name = "{}_update_data_lambda_function".format(env)
+    client = boto3.client("lambda")
+    body = json.dumps(data)
+    event = {"body": body}
+
+    client.invoke(FunctionName=name, Payload=json.dumps(event))
+
+
+def update_data_to_database(data, env):
+    (
+        df_companies,
+        df_time_periods,
+        df_scenarios,
+        df_metrics,
+        df_scenario_metrics,
+        df_currencies,
+    ) = get_updating_dataframes_from_lists(data)
+
+    logger.warning("==========Company===============================")
+    df_companies.show()
+    companies_updatables = df_companies.rdd.map(lambda x: x.asDict()).collect()
+
+    logger.warning("==========Existing Metrics===============================")
+    result_metrics_df = df_metrics.collect()
+
+    new_metrics = []
+    existing_metrics = []
+    for row in result_metrics_df:
+        if validate_existing_metric_from_dataframe(env, row.id):
+            existing_metrics.append(row)
+        else:
+            new_metrics.append(row)
+    df_existing_metrics = spark.createDataFrame(existing_metrics, schema=metric_schema)
+    df_new_metrics = spark.createDataFrame(new_metrics, schema=metric_schema)
+    df_existing_metrics.show()
+
+    metrics_updatables = df_existing_metrics.rdd.map(lambda x: x.asDict()).collect()
+    data_to_update = {"companies": companies_updatables, "metrics": metrics_updatables}
+    call_lambda_to_update(env, data_to_update)
+
+    if (
+        df_new_metrics.collect() != []
```

Comment:
You could use df_new_metrics.collect() only, if <empty_list> is equals to if false

Reply:
I used this.